jane/doc/parallelism-tutorial.md

```diff
@@ -77,36 +197,71 @@ let%expect_test "add4 in parallel" =
 This creates a work-stealing scheduler, along with a _monitor_ to manage
 exceptions. Then it tells the scheduler to run the `test_add4` function before
 shutting down the scheduler. (Naturally, a real program will want to keep the
-monitor and scheduler around longer!) To test using the sequential scheduler
-instead, we would simply replace `Parallel_scheduler_work_stealing` with
-`Parallel.Scheduler.Sequential`.
+monitor and scheduler around longer!)
+
+You can also use the `parallel` library's own `Parallel.Scheduler.Sequential`,
+which simply runs everything on the primary domain. This is handy for testing or
+debugging when you want to eliminate nondeterminism. To do so, simply replace
+`Parallel_scheduler_work_stealing` with `Parallel.Scheduler.Sequential`.
+
+Now for something more substantial. Suppose we're working with binary trees, and
+we want to take an average over all the values in the tree. Here's a basic
+implementation:
+
+<a id="code-average"></a>
+```ocaml
+module Tree = struct
+  type 'a t =
+    | Leaf of 'a
+    | Node of 'a t * 'a t
+end
+
+let average (tree : float Tree.t) =
+  let rec total tree : total:float * count:int =
```

Comment:
```suggestion
  let rec total tree : (total:float * count:int) =
```

The parens are redundant, of course, but given the other `:` on the line, I think it makes the code easier to read.